common/utils/include/claragenomics/utils/allocator.hpp

```diff
@@ -226,7 +226,7 @@ class CachingDeviceAllocator
     void deallocate(pointer p, std::size_t n, cudaStream_t stream = 0)
     {
         // deallocate should not throw execeptions which is why CGA_CU_CHECK_ERR is not used.
```

Comment:
this comment could be removed now.

Reply:
@ahehn-nv @akkamesh:
@r-mafi raised a good point here. We probably should not throw here, but on the other hand we need a way to let the developer/user know that something went wrong. Any idea how to solve this? Having the application crash does sound a extreme, but if deallocation does actually fail...

Reply:
@mimaric in the article you posted (https://akrzemi1.wordpress.com/2011/09/21/destructors-that-throw/), it mentions: 
"...For the end, note that while no destructor in C++ Standard Library is allowed to throw exceptions, destructors are allowed to std::terminate the program in case of an error..."
CGA_CU_CHECK_ERR calls `assert(false)` and `std::abort();` so IMO it should be OK.

Reply:
Yes, you are right - given our current "fail hard" error handling this is no problem.
However, what if we try to change that sometime in the future? Then we need to go through every destructor and fix that again.
I'm not sure if this is a purely hypothetical problem, since it is unlikely that we will change this. Nevertheless, an easy future-prove method would be to introduce a `CGA_CU_ABORT_ON_ERR()`, which does the same as `CGA_CU_CHECK_ERR()` but allows us to distinguish the two if we ever need to.
Any opinions on that?

Reply:
Yes, I like the idea of having `CGA_CU_ABORT_ON_ERR()`. I'll implement it and update this PR

Reply:
done